File: prediction-engine/test_db_connection.py
"""
test_db_connection.py
Script de test de connexion PostgreSQL pour le moteur de prédiction Python
Usage: python test_db_connection.py
"""

import os
import sys
from dotenv import load_dotenv

load_dotenv()

DATABASE_URL = os.getenv("DATABASE_URL")

def test_connection():
    print("\n🔌 Test de connexion PostgreSQL (DevMind Prediction Engine)\n")
    print(f"DATABASE_URL: {'✅ Définie' if DATABASE_URL else '❌ Non définie'}")

    if not DATABASE_URL:
        print("\n❌ DATABASE_URL non définie. Créez votre fichier .env à partir de .env.example\n")
        sys.exit(1)

    try:
        import psycopg2
    except ImportError:
        print("\n❌ psycopg2 non installé. Exécutez : pip install -r requirements.txt\n")
        sys.exit(1)

    try:
        conn = psycopg2.connect(DATABASE_URL, sslmode='require')
        cur = conn.cursor()

        print("\n✅ Connexion réussie à PostgreSQL !\n")

        # Test 1 : Version PostgreSQL
        cur.execute("SELECT version()")
        version = cur.fetchone()[0]
        print(f"📦 Version PostgreSQL: {' '.join(version.split()[:2])}")

        # Test 2 : Lister les tables
        cur.execute("""
            SELECT table_name
            FROM information_schema.tables
            WHERE table_schema = 'public'
            ORDER BY table_name;
        """)
        tables = [row[0] for row in cur.fetchall()]

        if not tables:
            print("\n⚠️  Aucune table trouvée — exécutez la migration Node.js d'abord :")
            print("   cd ../backend && npm run migrate\n")
        else:
            print(f"\n📋 Tables trouvées ({len(tables)}) :")
            for t in tables:
                print(f"   - {t}")

            # Test 3 : Tables critiques pour le moteur Python
            expected = [
                'matches', 'predictions', 'markets', 'ai_models',
                'leagues', 'teams', 'coupons', 'coupon_predictions',
                'prediction_results', 'risk_factors'
            ]
            missing = [t for t in expected if t not in tables]
            if missing:
                print(f"\n⚠️  Tables manquantes : {', '.join(missing)}")
            else:
                print("\n✅ Toutes les tables requises pour le moteur Python sont présentes !")

            # Test 4 : Modèles IA disponibles
            cur.execute("SELECT name, version FROM ai_models WHERE is_active = TRUE")
            models = cur.fetchall()
            if models:
                print(f"\n🤖 Modèles IA actifs ({len(models)}) :")
                for name, version in models:
                    print(f"   - {name} v{version}")

            # Test 5 : Marchés disponibles
            cur.execute("SELECT code, name FROM markets WHERE is_active = TRUE")
            markets = cur.fetchall()
            if markets:
                print(f"\n🎯 Marchés actifs ({len(markets)}) :")
                for code, name in markets:
                    print(f"   - {code}: {name}")

        cur.close()
        conn.close()
        print("\n✅ Tous les tests ont réussi !\n")

    except psycopg2.OperationalError as e:
        print(f"\n❌ Erreur de connexion: {e}")
        print("\nVérifiez :")
        print("  1. Que DATABASE_URL est correctement définie dans .env")
        print("  2. Que votre base de données Neon.tech / Supabase est accessible")
        print("  3. Que les options SSL sont correctes\n")
        sys.exit(1)
    except Exception as e:
        print(f"\n❌ Erreur inattendue: {e}\n")
        sys.exit(1)


if __name__ == "__main__":
    test_connection()

File: backend/test-db-connection.js
/**
 * test-db-connection.js
 * Script de test de connexion à la base de données PostgreSQL
 * Usage: node test-db-connection.js
 */

const path = require('path');
require('dotenv').config({ path: path.join(__dirname, '.env') });

const { Pool } = require('pg');

async function testConnection() {
  console.log('\n🔌 Test de connexion PostgreSQL (DevMind Bot Backend)\n');
  console.log(`DATABASE_URL: ${process.env.DATABASE_URL ? '✅ Définie' : '❌ Non définie'}`);

  if (!process.env.DATABASE_URL) {
    console.error('\n❌ DATABASE_URL non définie. Créez votre fichier .env à partir de .env.example\n');
    process.exit(1);
  }

  const pool = new Pool({
    connectionString: process.env.DATABASE_URL,
    ssl: { rejectUnauthorized: false },
    connectionTimeoutMillis: 15000,
  });

  try {
    const client = await pool.connect();
    console.log('\n✅ Connexion réussie à PostgreSQL !\n');

    // Test 1 : Version PostgreSQL
    const versionResult = await client.query('SELECT version()');
    console.log('📦 Version PostgreSQL:', versionResult.rows[0].version.split(' ').slice(0, 2).join(' '));

    // Test 2 : Lister les tables existantes
    const tablesResult = await client.query(`
      SELECT table_name
      FROM information_schema.tables
      WHERE table_schema = 'public'
      ORDER BY table_name;
    `);

    if (tablesResult.rows.length === 0) {
      console.log('\n⚠️  Aucune table trouvée — la migration n\'a pas encore été exécutée.');
      console.log('   Exécutez : npm run migrate\n');
    } else {
      console.log(`\n📋 Tables trouvées (${tablesResult.rows.length}) :`);
      tablesResult.rows.forEach(row => {
        console.log(`   - ${row.table_name}`);
      });

      // Test 3 : Vérifier les tables critiques
      const expectedTables = [
        'users', 'subscriptions', 'payments', 'matches', 'leagues',
        'teams', 'predictions', 'markets', 'coupons', 'coupon_predictions',
        'prediction_results', 'performance_stats', 'risk_factors',
        'ai_models', 'api_quota_tracker'
      ];

      const existingNames = tablesResult.rows.map(r => r.table_name);
      const missing = expectedTables.filter(t => !existingNames.includes(t));

      if (missing.length === 0) {
        console.log('\n✅ Toutes les tables requises sont présentes !');
      } else {
        console.log('\n⚠️  Tables manquantes :', missing.join(', '));
        console.log('   Exécutez : npm run migrate');
      }

      // Test 4 : Vérifier les seeds
      const modelsResult = await client.query('SELECT name, version FROM ai_models');
      if (modelsResult.rows.length > 0) {
        console.log(`\n🤖 Modèles IA enregistrés (${modelsResult.rows.length}) :`);
        modelsResult.rows.forEach(m => console.log(`   - ${m.name} v${m.version}`));
      }

      const leaguesResult = await client.query('SELECT name, country FROM leagues WHERE covered = TRUE');
      if (leaguesResult.rows.length > 0) {
        console.log(`\n⚽ Championnats couverts V1 (${leaguesResult.rows.length}) :`);
        leaguesResult.rows.forEach(l => console.log(`   - ${l.name} (${l.country})`));
      }

      const marketsResult = await client.query('SELECT code, name FROM markets');
      if (marketsResult.rows.length > 0) {
        console.log(`\n🎯 Marchés de paris (${marketsResult.rows.length}) :`);
        marketsResult.rows.forEach(m => console.log(`   - ${m.code}: ${m.name}`));
      }
    }

    client.release();
    console.log('\n✅ Tous les tests ont réussi !\n');

  } catch (err) {
    console.error('\n❌ Erreur de connexion:', err.message);
    console.error('\nVérifiez :');
    console.error('  1. Que DATABASE_URL est correctement définie dans .env');
    console.error('  2. Que votre base de données Neon.tech / Supabase est accessible');
    console.error('  3. Que les options SSL sont correctes\n');
    process.exit(1);
  } finally {
    await pool.end();
  }
}

testConnection();
